.github/workflows/ci.yml

```
@@ -0,0 +1,152 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+env:
+  CARGO_TERM_COLOR: always
+  RUST_BACKTRACE: 1
+
+permissions:
+  contents: read
+
+jobs:
+  format:
+    name: Format
+    runs-on:
+      group: arc-public-large-amd64-runner
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          components: rustfmt
+
+      - name: Check formatting
+        run: cargo fmt --all -- --check
+
+  lint:
+    name: Lint
+    runs-on:
+      group: arc-public-large-amd64-runner
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          components: clippy
+
+      - name: Cache dependencies
+        uses: Swatinem/rust-cache@v2
+
+      - name: Run Clippy
+        run: cargo clippy --all-targets --all-features --
+
+  fast-build:
+    name: Build
+    runs-on:
+      group: arc-public-large-amd64-runner
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@stable
+
+      - name: Cache dependencies
+        uses: Swatinem/rust-cache@v2
+
+      - name: Build
+        run: SQLX_OFFLINE=true cargo build --verbose
+
+      - name: Build release
+        run: SQLX_OFFLINE=true cargo build --release --verbose
+
+  test:
+    name: Tests
+    runs-on:
+      group: arc-public-large-amd64-runner
```

@Takaros999 I suggest use 2xlarge because of OOM
```suggestion
      group: arc-public-2xlarge-amd64-runner
```